pycaret/loggers/dagshub_logger.py

```diff
@@ -0,0 +1,94 @@
+import os
+from pathlib import Path
+
+from pycaret.loggers.mlflow_logger import MlflowLogger
+
+try:
+    import dagshub
+    import mlflow
+    from dagshub.upload import Repo
+except ImportError:
+    dagshub = None
+
+
+class DagshubLogger(MlflowLogger):
+    def __init__(self, remote=None) -> None:
+        super().__init__()
+        if dagshub is None:
+            raise ImportError(
+                "DagshubLogger requires dagshub. Install using `pip install dagshub`"
+            )
+        # check token exist or not:
+        token = dagshub.auth.get_token()
+        os.environ["MLFLOW_TRACKING_USERNAME"] = token
+        os.environ["MLFLOW_TRACKING_PASSWORD"] = token
+
+        # Check mlflow environment variable is set:
+        if not remote or "dagshub" not in os.getenv("MLFLOW_TRACKING_URI"):
+            prompt_in = input(
+                "Please insert your repository owner_name/repo_name:"
+            ).split("/")
+            assert (
+                len(prompt_in) == 2
+            ), f"Invalid input, should be owner_name/repo_name, but get {prompt_in} instead"
+
+            dagshub.init(repo_name=prompt_in[1], repo_owner=prompt_in[0])
+            remote = os.getenv("MLFLOW_TRACKING_URI")
```

Comment:
Can we add a path to do this programatically? eg. passing `owner_name/repo_name` as a string argument.

Reply:
We did discuss this internally! We noted that it would require us to update the `setup` function - but this variable would only be used if we were logging using DagsHub, so it felt a bit unwieldy. WDYT?

Reply:
It is possible to pass the logger object directly as well, so there's that escape hatch. I don't think we need to modify the `setup` function here - I am fine with `input` being the primary mode of operation, but it shouldn't be the only one :)

Reply:
@jinensetpal let's add it as a var for the DagsHub logger object with default value = None.
If it's None, then ask for input.

Reply:
Perfect